server/src/main/java/fr/uga/l3miage/tp4/repositorys/CandidateRepository.java

```diff
@@ -0,0 +1,21 @@
+package fr.uga.l3miage.tp4.repositorys;
+import fr.uga.l3miage.tp4.enums.TestCenterCode;
+import fr.uga.l3miage.tp4.models.CandidateEntity;
+import fr.uga.l3miage.tp4.models.CandidateEvaluationGridEntity;
+import fr.uga.l3miage.tp4.models.TestCenterEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.*;
+
+
+import java.time.LocalDate;
+import java.util.Set;
+
+@Repository
+public interface CandidateRepository extends JpaRepository<CandidateEntity, Long> {
+    Set<TestCenterEntity> findAllByCodeEquals(TestCenterCode GRE);
+    Set<CandidateEntity> findAllByGradeLessThanEqual(double grade);
```

Comment:
non ton candidat n'a pas la note directement, il faut la récupérer dans la grid ! 
```suggestion
    Set<CandidateEntity> findAllByCandidateEvaluationGridEntitiesGradeLessThanEqual(Double grade);
```